presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewQueryOptimizer.java

```diff
@@ -71,418 +86,748 @@
 import static com.facebook.presto.sql.ExpressionUtils.removeGroupingElementPrefix;
 import static com.facebook.presto.sql.ExpressionUtils.removeSingleColumnPrefix;
 import static com.facebook.presto.sql.ExpressionUtils.removeSortItemPrefix;
+import static com.facebook.presto.sql.ParsingUtil.createParsingOptions;
 import static com.facebook.presto.sql.analyzer.MaterializedViewInformationExtractor.MaterializedViewInfo;
 import static com.facebook.presto.sql.analyzer.SemanticErrorCode.MISSING_TABLE;
 import static com.facebook.presto.sql.analyzer.SemanticErrorCode.NOT_SUPPORTED;
 import static com.facebook.presto.sql.relational.Expressions.call;
 import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
 
+/**
+ * Rewrites a query by optimizing any "leaf queries" (queries which are made from a table,
+ * e.g. SELECT foo FROM t1 [ AS bar]) which can be optimized with an existing materialized view.
+ * A leaf query can be a subquery, but does not need to be. {@link MaterializedViewQueryOptimizer
+ * copies the input Query, handing off leaf queries to {@link QuerySpecificationRewriter#rewrite}
+ * for potential rewriting
+ */
 public class MaterializedViewQueryOptimizer
         extends AstVisitor<Node, Void>
 {
-    private static final Logger logger = Logger.get(MaterializedViewQueryOptimizer.class);
+    private static final Logger log = Logger.get(MaterializedViewQueryOptimizer.class);
 
     private static final QualifiedName MIN = QualifiedName.of("MIN");
     private static final QualifiedName MAX = QualifiedName.of("MAX");
     private static final QualifiedName SUM = QualifiedName.of("SUM");
     private static final QualifiedName COUNT = QualifiedName.of("COUNT");
     private static final Set<QualifiedName> SUPPORTED_FUNCTION_CALLS = ImmutableSet.of(MIN, MAX, SUM, COUNT);
 
+    private final LogicalRowExpressions logicalRowExpressions;
```

Comment:
Let's put this line back next to `RowExpressionDomainTranslator domainTranslator` minimize the change.